### PR TITLE
ci: Add support for ubuntu 22.04

### LINKED ISSUE #4901: ci: Add support for ubuntu 22.04

### PR DESCRIPTION
**Pull request by GabyCT**
This PR adds the suppor for ubuntu 22.04 in order to have it
in our kata CI.

Fixes #4901
Depends-on: github.com/https://github.com/kata-containers/kata-containers/pull/4397

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/nydus/nydus_tests.sh [hunk 1]
```diff
@@ -28,6 +29,11 @@ containerd_config_backup="/tmp/containerd.config.toml"
 # test image for container
 IMAGE="${IMAGE:-ghcr.io/dragonflyoss/image-service/alpine:nydus-latest}"
 
+if [ "${NAME}" == "Ubuntu" ] && [ "$(echo "${VERSION_ID} >= 22.04" | bc -q)" == "1" ]; then
+	echo "Skyp nydus tests as they are not working"
```
Comment (amshinde) [inline integration/nydus/nydus_tests.sh:33]: Typo for "Skip" here

File: integration/kubernetes/k8s-hugepages.bats [hunk 2]
```diff
@@ -6,8 +6,10 @@
 
 load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
+source "/etc/os-release" || source "/usr/lib/os-release"
 
 setup() {
+	[ "${NAME}" == "Ubuntu" ] && [ "$(echo "${VERSION_ID} >= 22.04" ] && skip "test not working"
```
Comment (amshinde) [inline integration/kubernetes/k8s-hugepages.bats:12]: Can you specify the test name in the message here ?

File: functional/vcpus/default_vcpus_test.sh [hunk 3]
```diff
@@ -29,6 +30,11 @@ if [ "$TEST_INITRD" == "yes" ]; then
 	exit 0
 fi
 
+if [ "${NAME}" == "Ubuntu" ] && [ "$(echo "${VERSION_ID} >= 22.04" | bc -q)" == "1" ]; then
+	echo "Skip vcpu test is not working"
```
Comment (amshinde) [inline functional/vcpus/default_vcpus_test.sh:34]: Whats the error you are seeing with vcpu tests?

Comment (GabyCT) [reply functional/vcpus/default_vcpus_test.sh:34]: @amshinde 
12:45:42 Running ctr integration tests with vcpus
12:45:43 quay.io/prometheus/busybox:latest: resolving      |[32m[0m--------------------------------------| 
12:45:43 elapsed: 0.1 s                     total:   0.0 B (0.0 B/s)                                         
12:45:43 quay.io/prometheus/busybox:latest:                                                resolved       |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 index-sha256:918fd39802562ff02d4232970ee50a11f7884a2699adfefdb500fa61e466c0cc:    done           |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 manifest-sha256:6594015d3ec4a8fa22c96d419491c2064462e409dd59b68101297de59f83f55d: done           |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 layer-sha256:7fe869475116a421aa5ff34a0dd1c4015e4b04f2b0b86682cea21b9695ff77f5:    done           |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 config-sha256:342b02f8cb5f577c5868ed34f8d8901be614b67cf6e82817fadeb0e612292c80:   done           |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 layer-sha256:19d511225f94f9b5cbf3836eb02b5273c01b95da50735742560e3e45b8c8bfcc:    done           |[32m++++++++++++++++++++++++++++++++++++++[0m| 
12:45:43 elapsed: 0.2 s                                                                    total:   0.0 B (0.0 B/s)                                         
12:45:43 unpacking linux/amd64 sha256:918fd39802562ff02d4232970ee50a11f7884a2699adfefdb500fa61e466c0cc...
12:45:43 done: 11.468708ms	
12:45:43 ctr: failed to create shim: write /sys/fs/cgroup/vc/cgroup.procs: device or resource busy: unknown

**Comment by liubin** [comment]
@GabyCT Thank you, maybe creating one issue and writing the issue before the skipping line will help to track them.

**Comment by GabyCT** [comment]
@liubin changes applied

**Review by liubin — APPROVED** [review]

**Comment by liubin** [comment]
Thank you @GabyCT

**Pull request merged**